lib/app_profiler/railtie.rb

```diff
@@ -40,6 +40,7 @@ class Railtie < Rails::Railtie
       AppProfiler.profile_enqueue_success = app.config.app_profiler.profile_enqueue_success
       AppProfiler.profile_enqueue_failure = app.config.app_profiler.profile_enqueue_failure
       AppProfiler.after_process_queue = app.config.app_profiler.after_process_queue
+      AppProfiler.backend = app.config.app_profiler.profiler_backend || :stackprof
```

Comment:
```suggestion
      unless AppProfiler.running?
        AppProfiler.backend = app.config.app_profiler.profiler_backend || :stackprof
      end
```
For boot time profiles, otherwise it crashes.

Reply:
@gmcgibbon why does it crash? maybe we need to also set this constant in lib/app_profiler.rb?

Reply:
The first time we reference the backend, if it is nil, it should default to stackprof already:

https://github.com/Shopify/app_profiler/pull/101/files#diff-bbd7626aa479b35f000a50b4e1c7d1f660c1eaaf1fa2c2ed71a1f12775a239afR130

So i'd like to see the backtrace for the boot time crash to ensure safety if this is not adequate

Reply:
or, is the crash happening because it tries to update the backend while profiling is already happening? If so, i'd like to make that impossible rather than just guard for it here.